pkg/scale/decode.go

```diff
@@ -470,7 +541,7 @@ func (ds *decodeState) decodeBool(dstv reflect.Value) (err error) {
 	case 0x01:
 		b = true
 	default:
-		err = fmt.Errorf("could not decode invalid bool")
+		err = fmt.Errorf("%w: %s", errDecodeBool, err)
```

Comment:
`err == nil` in this case. Why  do we need to print `nil`?

Reply:
and also export `errDecodeBool` I guess if it's now a sentinel error :thinking: 

```suggestion
		err = errDecodeBool
```